airbyte-integrations/connectors/source-mysql/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlJdbcPartitionFactory.kt

```diff
@@ -158,7 +157,10 @@ class MysqlJdbcPartitionFactory(
         val stream: Stream = streamFeedBootstrap.feed
         val streamState: DefaultJdbcStreamState = streamState(streamFeedBootstrap)
         val opaqueStateValue: OpaqueStateValue =
-            streamFeedBootstrap.currentState ?: return coldStart(streamState)
+            when (streamFeedBootstrap.currentState?.isEmpty) {
+                false -> streamFeedBootstrap.currentState!!
+                else -> return coldStart(streamState)
+            }
```

Comment:
This fixes a bug:
When sync that includes a number of feeds with some that haven't started yet, during checkpoint those feeds will have a stream state of `{}`.
Stopping and restarting the sync need to treat them as a cold start. before this fix those `{}` feeds didn't do a snapshot.